v2/client_test.go

```diff
@@ -84,6 +96,16 @@ func doHTTP(t *testing.T, name string, checks httpChecks, reaction httpReaction)
 			return nil, walkingGhostErr
 		}
 
+		if len(checks.headers) > 0 {
```

Comment:
technically the "if" is needed since the `range` on the next line will return an empty enum if there's nothing in "headers".